src/persist-client/src/read.rs

```diff
@@ -228,19 +303,22 @@ where
     T: Timestamp + Lattice + Codec64,
     D: Semigroup + Codec64 + Send + Sync,
 {
-    async fn new(handle: ReadHandle<K, V, T, D>, as_of: Antichain<T>) -> Self {
-        let mut ret = Listen {
-            handle,
-            since: as_of.clone(),
-            frontier: as_of.clone(),
-            as_of,
-        };
-
+    async fn new(mut handle: ReadHandle<K, V, T, D>, as_of: Antichain<T>) -> Self {
+        let since = as_of.clone();
         // This listen only needs to distinguish things after its frontier
         // (initially as_of although the frontier is inclusive and the as_of
         // isn't). Be a good citizen and downgrade early.
-        ret.handle.downgrade_since(&ret.since).await;
-        ret
+        handle.downgrade_since(&since).await;
+
+        let fetcher = handle.clone().await.batch_fetcher().await;
```

Comment:
hmm this is registering and expiring a ReaderId in durable state. I think it just wants to call the BatchFetcher::new method directly, instead

edit after reading below: better yet, this also just want to call a freestanding `fetch_batch` directly